src/odemis/driver/xt_client.py

```diff
@@ -349,20 +468,579 @@ def set_autofocusing(self, channel, state):
+        fwd_info = self.parent.fwd_info()
+        axes_def = {
+            "z": model.Axis(unit=fwd_info["unit"], range=fwd_info["range"]),
+        }
+
+        model.Actuator.__init__(self, name, role, parent=parent, axes=axes_def, **kwargs)
+
+        # will take care of executing axis move asynchronously
+        self._executor = CancellableThreadPoolExecutor(max_workers=1)  # one task at a time
+
+        # RO, as to modify it the server must use .moveRel() or .moveAbs()
+        self.position = model.VigilantAttribute({}, unit="m", readonly=True)
+        self._updatePosition()
+
+        # Refresh regularly the position
+        self._pos_poll = util.RepeatingTimer(5, self._refreshPosition, "Position polling")
+        self._pos_poll.start()
+
+    def _updatePosition(self):
+        """
+        update the position VA
+        """
+        z = self.parent.get_free_working_distance()
+        self.position._set_value({"z": z}, force_write=True)
+
+    def _refreshPosition(self):
+        """
+        Called regularly to update the current position
+        """
+        # We don't use the VA setters, to avoid sending back to the hardware a
+        # set request
+        logging.debug("Updating SEM stage position")
+        try:
+            self._updatePosition()
+        except Exception:
+            logging.exception("Unexpected failure when updating position")
+
+    def _doMoveRel(self, foc):
+        """
+        move by foc
+        foc (float): relative change in m
+        """
+        try:
+            foc += self.parent.get_free_working_distance()
+            self.parent.set_free_working_distance(foc)
+        finally:
+            # Update the position, even if the move didn't entirely succeed
+            self._updatePosition()
+
+    def _doMoveAbs(self, foc):
+        """
+        move to pos
+        foc (float): unit m
+        """
+        try:
+            self.parent.set_free_working_distance(foc)
+        finally:
+            # Update the position, even if the move didn't entirely succeed
+            self._updatePosition()
+
+    @isasync
+    def moveRel(self, shift):
+        """
+        shift (dict): shift in m
+        """
+        if not shift:
+            return model.InstantaneousFuture()
+        self._checkMoveRel(shift)
+
+        foc = shift["z"]
+        f = self._executor.submit(self._doMoveRel, foc)
+        return f
+
+    @isasync
+    def moveAbs(self, pos):
+        """
+        pos (dict): pos in m
+        """
+        if not pos:
+            return model.InstantaneousFuture()
+        self._checkMoveAbs(pos)
+
+        foc = pos["z"]
+        f = self._executor.submit(self._doMoveAbs, foc)
+        return f
+
+    def stop(self, axes=None):
+        """
+        Stop the last command
+        """
+        # Empty the queue (and already stop the stage if a future is running)
+        self._executor.cancel()
+        logging.debug("Stopping all axes: %s", ", ".join(self.axes))
```

Comment:
This message is not really accurate, as there is only 1 axis, and there is no way to stop the current move (which is pretty much instantaneous anyway). I'd suggest something like "Cancelled all ebeam focus moves".